### PR TITLE
[content] website testing methodology

### PR DESCRIPTION
**Pull request by markjmeier**
@robingustafsson First draft of a website/webapp methodology/guide.  My goal was to be more specific on the steps.  It does repeat a lot of concepts from other articles.

### CODE REVIEW

File: jekyll/_v4/testing-methodologies/website-testing-methodology.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+## Other things to think about for e-commerce specific flows
+
+- You likely have various systems connected to the front end you are testing.
+  - Payments
+    - In most cases you should not be testing your payment provider (unless you have explicit permission)
+  - Email confirmations
+    - Do any actions generate emails?  Make sure they are handled appropraitely.
+    - Avoid sending emails to a non existent mailbox and avoid sending emails to a single person. Try to disconnect this, if possible.
+  - Inventory
+    - Are the SKUs/products being tested have enough inventory?
```
Comment (robingustafsson) [inline jekyll/_v4/testing-methodologies/website-testing-methodology.md]: Should the "Are" be "Do" here? You obviously know better than me, but the sentence reads strange to me when it starts with "Are" and then "...have enough..".

File: jekyll/_v4/testing-methodologies/website-testing-methodology.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+</div>
+
+- For login, you'll probably want to include a CSV or JSON file containing usernames or passwords.
+  - Examples here: [Parameterizing from a JSON or CSV file]({{ site.baseurl}}/4.0/test-scripting/examples/#data-filesparameterization)
+- For CSRF tokens, you will need to: Identify the page where the token is created, save the response body of that page, find the token through the DOM or a regex match, save that to a variable, cocatenate the variable in future requests
+  - Examples here: [Data Correlation in your test script]({{ site.baseurl }}/4.0/test-scripting/examples/#correlation)
+- For other forms, such as a search or name fields, you can pull the data from an external source like you did with the login. You may prefer to generate the data inline if it doesn't matter what is submitted.  There are some good implementations of this on [stackoverflow](https://stackoverflow.com/questions/1349404/generate-random-string-characters-in-javascript) that you may want to utilize.
```
Comment (robingustafsson) [inline jekyll/_v4/testing-methodologies/website-testing-methodology.md]: Ah, this reminded me of [faker.js](https://github.com/marak/Faker.js/). I'll put together an example how that can be used in k6 to generate fake data.

File: jekyll/_v4/testing-methodologies/website-testing-methodology.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+***
+
+<h1>Background</h1>
+
+If you have already read our [Getting Started Methodology]({{ site.baseurl }}/4.0/testing-methodologies/application-testing-methodology/) similar concepts are repeated here. This document will touch on some more specific aspects to help you start testing a real user journey of a website or webapp. Testing of these systems generally involve simulating real user behavior and complex actions in some logical flow or journey, for this reason we have combined this into a single document. As you start your testing we recommend that you _start small and iterate, iterate, iterate._
+
+
+## Preparations
+The most important step before any testing, here are things find answers to in order to improve your testing outcomes.
```
[the remark this answers is not in the document]
Comment (markjmeier) [reply jekyll/_v4/testing-methodologies/website-testing-methodology.md]: Yes - it is connected. But I think I can make it clearer to the reader.

File: jekyll/_v4/testing-methodologies/website-testing-methodology.md (hunk 1 above)
Comment (markjmeier) [reply jekyll/_v4/testing-methodologies/website-testing-methodology.md]: Yes - definitely weird / typo

**Comment by markjmeier** [comment]
Merged with Master

**Pull request merged**